effectful/ops/semantics.py

```diff
@@ -47,7 +47,7 @@ def apply(intp: Interpretation, op: Operation, *args, **kwargs) -> Any:
 
     """
     if op in intp:
-        return intp[op](*args, **kwargs)
+        return handler(intp)(intp[op])(*args, **kwargs)
```

Comment:
I think this should probably use `effectful.internals.runtime.interpreter` rather than `handler`, which could in some cases lead to a `coproduct` of the active interpretation with itself.